lib/grplatt.gi

```diff
@@ -3335,57 +3335,254 @@ local c,n,deg,ind,core,i,j,sum;
 
 end);
 
+
+# utility function: Find subgroup of $G$, containing $G'$ of index up to
+# limit which is normal in the normalizer of G in P.
+BindGlobal("BoundedIndexAbelianized",function(P,G,limit)
+local d,ind,i,ma,ab,a,p,b,c,e,n;
+  d:=DerivedSubgroup(G);
+  ind:=IndexNC(G,d);
+  if ind=1 then return G;
+  # derived index small enough
+  elif ind<=limit then return d;
+  elif IsPrimeInt(ind) then return d;fi;
+
+  # make a p group
+  ind:=List(Collected(Factors(ind)),x->x[1]^x[2]);
+  Sort(ind);
+  if Length(ind)>1 then 
+    i:=0;
+    while i<Length(ind) and ind[i+1]<=limit do
+      i:=i+1;
+    od;
```

Comment:
Isn't the above just the same as `i := PositionSorted(ind, limit);` ? (Ah well, maybe not if limit exceeds all entries in `ind`... then the current code sets `i` to `Length(ind)`, while `PositionSorted` would set it to `Length(ind)+1`.

Reply:
I think it would have to be `PositionSorted(ind,limit+1)` to get the first position where `ind` is larger. Still, I don't think the code is hard to read or slow in practice, so I would prefer to keep it boring.

Reply:
Changed.